module/reports/helper.js

```diff
@@ -0,0 +1,443 @@
+                                categories[category.name] = 1;
+                            }
+                        });
+                        categories['total'] = categories['total'] + project.categories.length;
+                    }
+
+                    if (project.status == CONSTANTS.common.COMPLETED_STATUS) {
+
+                        projectReport[CONSTANTS.common.COMPLETED_STATUS] = projectReport[CONSTANTS.common.COMPLETED_STATUS] + 1;
+
+                    } else if (project.status == CONSTANTS.common.INPROGRESS_STATUS) {
+
+                        var todayDate = moment(project.endDate, "DD.MM.YYYY");
+                        var endDate = moment().format();
+                        if (todayDate.diff(endDate, 'days') < 1) {
+                            projectReport['overdue'] = projectReport['overdue'] + 1;
+                        } else {
+                            projectReport[CONSTANTS.common.INPROGRESS_STATUS] = projectReport[CONSTANTS.common.INPROGRESS_STATUS] + 1;
+                        }
+
+                    } else if (project.status == CONSTANTS.common.NOT_STARTED_STATUS) {
+
+                        var todayDate = moment(project.endDate, "DD.MM.YYYY");
+                        var endDate = moment().format();
+                        if (todayDate.diff(endDate, 'days') < 1) {
+                            projectReport['overdue'] = projectReport['overdue'] + 1;
+                        } else {
+                            projectReport[CONSTANTS.common.NOT_STARTED_STATUS] = projectReport[CONSTANTS.common.NOT_STARTED_STATUS] + 1;
+                        }
+                    }
+
+
+
+                    if (project.taskReport) {
+                        let keys = Object.keys(project.taskReport);
+                        keys.map(key => {
+                            if (tasksReport[key]) {
+                                tasksReport[key] = tasksReport[key] + project.taskReport[key];
+                            } else {
+                                tasksReport[key] = project.taskReport[key];
+                            }
+                        });
+                    }
+
+                    await Promise.all(project.tasks.map(task => {
+                        var todayDate = moment(task.endDate, "DD.MM.YYYY");
+                        var endDate = moment().format();
+                        if (todayDate.diff(endDate, 'days') < 1) {
+                            if (tasksReport['overdue']) {
+                                tasksReport['overdue'] = tasksReport['overdue'] + 1;
+                            } else {
+                                tasksReport['overdue'] = 1;
+                            }
+                        }
+                    }));
+
+
+                }));
+
+                if (projectReport[CONSTANTS.common.INPROGRESS_STATUS]) {
+                    projectReport['onGoing'] = projectReport[CONSTANTS.common.INPROGRESS_STATUS];
```

Comment:
@rakeshSgr - Why we consider inprogress and they do as ongoing ?

Cc - @aman-tunerlabs

Reply:
Rajneesh requested